### PR TITLE
feat: add bulk upsert methods for all database entities

### PR DESCRIPTION
**Pull request by ben-kaufman**
This PR introduces bulk upsert operations for orders, CJIT entries, Blocktank info, onchain activities, lightning activities, and closed channels. All bulk operations use transactions and prepared statements for optimal performance.

**Blocktank Module:**
- Add `upsert_orders()` for bulk upsert of IBtOrder records
- Add `upsert_cjit_entries()` for bulk upsert of ICJitEntry records
- Add `upsert_info()` method (existing, now exposed in bindings)
- Refactor `upsert_order()` and `upsert_cjit_entry()` to use helper functions
- Extract SQL constants to `models.rs` for reusability
- Add helper functions `build_order_params()` and `build_cjit_params()` to reduce duplication

**Activity Module:**
- Add `upsert_onchain_activities()` for bulk upsert of OnchainActivity records
- Add `upsert_lightning_activities()` for bulk upsert of LightningActivity records
- Rename `insert_closed_channel()` to `upsert_closed_channel()` (uses INSERT OR REPLACE)
- Add `upsert_closed_channels()` for bulk operations

**FFI Bindings:**
- Export `upsert_orders()` in lib.rs
- Export `upsert_cjit_entries()` in lib.rs
- Export `upsert_info()` in lib.rs
- Export `upsert_onchain_activities()` in lib.rs
- Export `upsert_lightning_activities()` in lib.rs
- Export `upsert_closed_channels()` in lib.rs (single method already exported)

**Testing:**
- Add tests for all bulk upsert methods
- Add tests for empty input handling

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1320,6 +1347,62 @@ pub async fn blocktank_wipe_all() -> Result<(), BlocktankError> {
...
+        let guard = cell.lock().await;
+        let db = guard.blocktank_db.as_ref().ok_or(BlocktankError::ConnectionError {
+            error_details: "Database not initialized. Call init_db first.".to_string()
+        })?;
+        let external_info: rust_blocktank_client::IBtInfo = info.into();
+        db.upsert_info(&external_info).await
+    }).await.unwrap_or_else(|e| Err(BlocktankError::ConnectionError {
+        error_details: format!("Runtime error: {}", e)
+    }))
+}
```
Comment (ovitrif) [inline src/lib.rs]: is this basically just inserting?
Not sure `upsert` is very accurate of a name here, because we don't have an identifier… but then again, it's not really an `insert` if it replaces the existing `Info` in DB.

Maybe `setInfo` or `updateInfo`?! 🤔  I would prefer `set` because, since `Info` can be null, then when null we don't update, we set it.

**Review by ovitrif — APPROVED** [review]
LGTM, with a few comments for clarifications.

Untested, but since this is a lib upgrade, testing can be done afterwards.

File: src/lib.rs [hunk 2]
```diff
@@ -402,12 +402,39 @@ pub fn get_all_unique_tags() -> Result<Vec<String>, ActivityError> {
...
+    db.upsert_onchain_activities(&activities)
+}
+
+#[uniffi::export]
+pub fn upsert_lightning_activities(activities: Vec<LightningActivity>) -> Result<(), ActivityError> {
+    let mut guard = get_activity_db()?;
+    let db = guard.activity_db.as_mut().ok_or(ActivityError::ConnectionError {
+        error_details: "Database not initialized. Call init_db first.".to_string()
+    })?;
+    db.upsert_lightning_activities(&activities)
```
[the remark this answers is not in the document]
Comment (ben-kaufman) [reply src/lib.rs]: Sure, added `upsert_activities` in [6f144f4](https://github.com/synonymdev/bitkit-core/pull/46/commits/6f144f44bf949fdbace765d29eb41cbe2099f0aa)
